homeassistant/components/ourgroceries/todo.py

```diff
@@ -0,0 +1,125 @@
+"""A todo platform for OurGroceries."""
+
+import asyncio
+
+from homeassistant.components.todo import (
+    TodoItem,
+    TodoItemStatus,
+    TodoListEntity,
+    TodoListEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import OurGroceriesDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the OurGroceries todo platform config entry."""
+    coordinator: OurGroceriesDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        OurGroceriesTodoListEntity(coordinator, sl["id"], sl["name"])
+        for sl in coordinator.lists
+    )
+
+
+class OurGroceriesTodoListEntity(
+    CoordinatorEntity[OurGroceriesDataUpdateCoordinator], TodoListEntity
+):
+    """An OurGroceries TodoListEntity."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+    )
+
+    def __init__(
+        self,
+        coordinator: OurGroceriesDataUpdateCoordinator,
+        list_id: str,
+        list_name: str,
+    ) -> None:
+        """Initialize TodoistTodoListEntity."""
+        super().__init__(coordinator=coordinator)
+        self._list_id = list_id
+        self._attr_unique_id = list_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, list_id)},
+            manufacturer="OurGroceries",
+            name=list_name,
+            entry_type=DeviceEntryType.SERVICE,
```

Comment:
I think that maybe 1 device for 1 account would work best since you don't create 1 device (/service) per list for only 1 entity. would love to hear what you think

Reply:
The problem with one service is naming it - the best candidate would be the email used to login, but that ends up as a prefix to the entity name, and you can't even see the actual list name in the to-do lists screen.
Given no other todo integration creates devices, I really think we shouldn't either.

Reply:
oooh right, that's true.

Reply:
OK, I removed the devices and addressed the other comment